templates/repo/diff/conversation.tmpl

```diff
@@ -1,14 +1,21 @@
 {{$resolved := (index .comments 0).IsResolved}}
+{{$invalid := (index .comments 0).Invalidated}}
 {{$resolveDoer := (index .comments 0).ResolveDoer}}
 {{$isNotPending := (not (eq (index .comments 0).Review.Type 0))}}
+{{$referenceUrl := printf "%s#%s" $.Issue.Link (index .comments 0).HashTag}}
 <div class="conversation-holder" data-path="{{(index .comments 0).TreePath}}" data-side="{{if lt (index .comments 0).Line 0}}left{{else}}right{{end}}" data-idx="{{(index .comments 0).UnsignedLine}}">
 	{{if $resolved}}
 		<div class="ui attached header resolved-placeholder gt-df gt-ac gt-sb">
-			<div class="ui grey text">
+			<div class="ui grey text gt-df gt-ac gt-fw gt-gap-2">
 				{{svg "octicon-check" 16 "icon gt-mr-2"}}
 				<b>{{$resolveDoer.Name}}</b> {{$.locale.Tr "repo.issues.review.resolved_by"}}
+				{{if $invalid}}
```

Comment:
Hmm ... I still can't tell how it works. The logic looks like:

```
{{if $resolved}}
  {{if $invalid}}
  {{end}}
{{end}}
```

What if `$resolved=false` and `$invalid=true` (just a guess, not sure whether it would happen or whether it needs to be handled)

Reply:
I've added this as a comment in the template.